app/main.py

```diff
@@ -1,25 +1,70 @@
+from __future__ import annotations
 from abc import ABC
+from typing import Any
 
 
-class IntegerRange:
+class VisitorOutOfLimits(Exception):
     pass
 
 
+class IntegerRange:
+    def __init__(self, min_amount: int, max_amount: int) -> None:
+        self.min_amount = min_amount
+        self.max_amount = max_amount
+
+    def __set_name__(self, owner: SlideLimitationValidator, name: str) -> None:
+        self.public_name = name
+        self.protected_name = f"_{name}"
+
+    def __get__(self, instance: SlideLimitationValidator, owner: Any) -> int:
+        return getattr(instance, self.protected_name)
+
+    def __set__(self, instance: SlideLimitationValidator, value: int) -> None:
+        if not (self.min_amount <= value <= self.max_amount):
+            raise VisitorOutOfLimits("Noone gonna see this")
+        setattr(instance, self.protected_name, value)
+
+
 class Visitor:
-   pass
+    def __init__(self, name: str, age: int, weight: int, height: int) -> None:
+        self.name = name
+        self.age = age
+        self.weight = weight
+        self.height = height
 
 
 class SlideLimitationValidator(ABC):
-    pass
+    def __init__(self, age: int, weight: int, height: int) -> None:
+        self.age = age
+        self.weight = weight
+        self.height = height
 
 
 class ChildrenSlideLimitationValidator(SlideLimitationValidator):
-    pass
+    age = IntegerRange(4, 14)
+    height = IntegerRange(80, 120)
+    weight = IntegerRange(20, 50)
 
 
 class AdultSlideLimitationValidator(SlideLimitationValidator):
-    pass
+    age = IntegerRange(14, 60)
+    height = IntegerRange(120, 220)
+    weight = IntegerRange(50, 120)
 
 
 class Slide:
-    pass
+    def __init__(self, name: str, limitation_class: callable) -> None:
```

Comment:
limitation_class has  SlideLimitationValidator type

Reply:
but if I use this type, PyCharm underlines the following code and says, that it's type can be only callable
![image](https://github.com/user-attachments/assets/37d13362-06d3-4b62-9a7b-f6ebcec8a97b)

Reply:
Seems that there are some problem with Python version and Pycharm version, you can just push it like SlideLimitationValidator, I'll approve it)